### ISSUE TITLE
Request: support "required" in JSON schema v4

### ISSUE THREAD

**Issue by braco** [litixsoft/lx-valid#4]
This is one of the core features: http://json-schema.org/examples.html

```
{
  type: "object"
  required: ['foo']
  properties:
    foo:
      type: "string"
}
```

**Comment by 4kochi** [comment]
This is right. But in this point lx-valid differ from the json-schema.org rules. In our lib you can set the required attribute on every property. See https://github.com/litixsoft/lx-valid#example-for-types

**Comment by OliverFischer** [comment]
It would be desirable to reopen this issue again. We're heavily using json-schema across different programming langs (JS, Python, Java) and different parsers. As the specification says, required should be an array on the schema root level. As a simple workaround, we're removing the required array property on root level when working with lx-valid doing partial updates . Perhaps you're interested in a PR to be match the v4 specification?

**Comment by 4kochi** [comment]
Since version 1.2.0 lx-valid supports the array notation of required at root level.

**Comment by OliverFischer** [comment]
OK, may be I struggled when using the #getValidationFunction for update. It seems that only the attribute required is set to false on root level. Shouldn't the required array on root level be also removed/set to false?

```
if (options.isUpdate) {
                var i,
                    keys = Object.keys(schema.properties),
                    length = keys.length;

                for (i = 0; i < length; i++) {
                    if (!doc.hasOwnProperty(keys[i])) {
                        schema.properties[keys[i]].required = false;
                    }
                }
            }
```

Am I wrong ?

**Comment by 4kochi** [comment]
No, you are right. The `isUpdate` part of  `getValidationFunction()` has not been changed accordingly. I will check this. Thanks for the hint.

**Comment by OliverFischer** [comment]
Thanx in advance :-)

**Comment by 4kochi** [comment]
Released a new version 1.2.1 which should solve the problem.
